core/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -32,11 +32,11 @@
 public class ByteSizeValue implements Writeable {
 
     private final long size;
-    private final ByteSizeUnit sizeUnit;
+    private final ByteSizeUnit unit;
 
     public ByteSizeValue(StreamInput in) throws IOException {
         size = in.readVLong();
-        sizeUnit = ByteSizeUnit.BYTES;
+        unit = ByteSizeUnit.BYTES;
```

Comment:
Did you want to do the reading/writing of the unit here? That wouldn't make the PR too huge and it seems like what we want?